lib/types.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const conversions = require("webidl-conversions");
+
+const utils = require("./utils");
+
+module.exports.generateTypeConversion = function (name, idlType, argAttrs, customTypes, opts) {
+  opts = opts || {};
+  const requires = {};
+  const handleNullable = Boolean(opts.handleNullable);
+  let str = "";
+
+
+  if (handleNullable && idlType.nullable) {
+    str += `
+  if (${name} === null || ${name} === undefined) {
+    ${name} = null;
+  } else {`;
+  }
+
+  if (idlType.generic === "sequence") {
+    str += `
+  if (typeof ${name} !== "object" || typeof ${name}[Symbol.iterator] !== "function") {
+    throw new TypeError("The value provided is not an iterable object");
+  } else {
+    const V = [];
+    const tmp = ${name};
+    for (let nextItem of tmp) {`;
+
+    const conv = exports.generateTypeConversion("nextItem", idlType.idlType, [], customTypes, {
+      handleNullable: true
+    });
+    Object.assign(requires, conv.requires);
+    str += conv.body;
+
+    str += `
+      V.push(nextItem);
+    }
+    ${name} = V;
+  }`;
+  } else if (conversions[idlType.idlType] || customTypes.has(idlType.idlType)) {
+    const enforceRange = utils.getExtAttr(argAttrs, "EnforceRange");
+    const clamp = utils.getExtAttr(argAttrs, "Clamp");
+    const treatNullAs = utils.getExtAttr(argAttrs, "TreatNullAs");
+
+    let optString = "";
+    if (clamp) {
+      optString = `, { clamp: true }`;
+    } else if (enforceRange) {
+      optString = `, { enforceRange: true }`;
+    } else if (treatNullAs && treatNullAs.rhs.value === "EmptyString") {
+      optString = `, { treatNullAsEmptyString: true }`;
+    }
+    let conversionFn = null;
+    if (conversions[idlType.idlType]) {
+      conversionFn = `conversions["${idlType.idlType}"]`;
+    } else {
+      requires[`convert${idlType.idlType}`] = `require("./${idlType.idlType}").convert`;
+      conversionFn = `convert${idlType.idlType}`;
+    }
+    if (idlType.array) {
+      str += `
+  for (let i = 0; i < ${name}.length; ++i) {
+    ${name}[i] = ${conversionFn}(${name}[i]${optString});
+  }`;
+    } else {
+      str += `
+  ${name} = ${conversionFn}(${name}${optString});`;
+    }
+  } else {
+    str += `
+  ${name} = utils.tryImplForWrapper(${name});`;
+  }
+
+  if (handleNullable && idlType.nullable) {
+    str += `
+  }`;
+  }
+
+  return {
+    requires,
+    body: str
+  };
+};
+
+module.exports.canHandleType = function (idlType, customTypes) {
+  return idlType.generic === "sequence" || conversions[idlType.idlType] || customTypes.has(idlType.idlType);
```

Comment:
`idlType.idlType in conversions` seems a bit nicer since it guarantees the return value will always be a boolean.